service/spawn.go

```diff
@@ -317,6 +339,16 @@ func (uis *UIServer) modifySpawnHost(w http.ResponseWriter, r *http.Request) {
 			uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrap(err, "Error extending host expiration time"))
 			return
 		}
+		// make host expirable
+		if h.NoExpiration {
```

Comment:
Is this check inverted?

Reply:
No--if the host is currently set to non-expirable and we want to make it expirable (which is true if we get past the first if statement) then we need to set NeverExpire in the host document back to false. I'll improve the comment and iron out the details in staging.